applications/NXstress.nxdl.xml

```diff
@@ -0,0 +1,1115 @@
+            </doc>
+            <field name="name" type="NX_CHAR" minOccurs="1" maxOccurs="1">
+                <doc>
+                    Descriptive name of sample
+                </doc>
+            </field>
+            <field name="chemical_formula" minOccurs="0" maxOccurs="1">
+		        <doc>
+			        The chemical formula specified using CIF conventions.
+			        Abbreviated version of CIF standard: 
+
+			        * Only recognized element symbols may be used.
+			        * Each element symbol is followed by a 'count' number. A count of '1' may be omitted.
+			        * A space or parenthesis must separate each cluster of (element symbol + count).
+			        * Where a group of elements is enclosed in parentheses, the multiplier for the 
+			          group must follow the closing parentheses. That is, all element and group 
+			          multipliers are assumed to be printed as subscripted numbers.
+			        * Unless the elements are ordered in a manner that corresponds to their chemical 
+			          structure, the order of the elements within any group or moiety depends on 
+			          whether or not carbon is present.
+			        * If carbon is present, the order should be: 
+
+			          - C, then H, then the other elements in alphabetical order of their symbol. 
+			          - If carbon is not present, the elements are listed purely in alphabetic order of their symbol. 
+
+			        * This is the *Hill* system used by Chemical Abstracts.
+		        </doc>
+	        </field>
+            <field name="temperature" type="NX_FLOAT" units="NX_TEMPERATURE" minOccurs="0" maxOccurs="1">
+		        <doc>Sample temperature. This could be a scanned variable</doc>
+		        <dimensions rank="anyRank">
+		                	<dim index="1" value="n_Temp"/><!-- could be any length -->
+		        </dimensions>
+	        </field>
+            <field name="stress_field" type="NX_FLOAT" units="NX_ANY" minOccurs="0" maxOccurs="1">
+		        <doc>Applied external stress field</doc>
+		        <dimensions>
+			            <dim index="1" value="n_sField"/><!-- could be any length -->
+		        </dimensions>
+                <attribute name="direction">
+        	        <enumeration>
+        		        <item value="x"/>
+        		        <item value="y"/>
+        		        <item value="z"/>
+        	        </enumeration>
+                </attribute>
+	        </field>
+            <group name="gauge_volume" type="NXparameters" minOccurs="0" maxOccurs="1">
+                <doc> The gauge volume can be described with the following parameters:
+                 .. figure:: stress/GV2.jpg
+                  :width: 70%
+                  :alt: Dimensions of the gauge volume.
+
+                  Parameters describing the gauge volume.
+                </doc>
+                <field name="x" type="NX_FLOAT" units="NX_LENGTH" minOccurs="0" maxOccurs="1">
+                    <doc>
+                        Length of the first diagonal.

```

Comment:
It isn't clear how these diagonals relate to the local coordinate frame

Reply:
Indeed. I'm guessing the volume was important, not really the orientation. Nevertheless it might be best to be precise. I'll make a drawing with the XYZ coordinate system in the bottom-left corner. It is inconvenient that the diagonals and the height is also called x, y and z ...

Reply:
I renamed the fields x, y and z to a, b and c. The gauge volume now has a local XYZ coordinate system with a depends_on. The base class `NXparameters` does not have it but I can add it in an application definition no?

Reply:
Yep, that's the standard practice